### ISSUE TITLE
Flashing problem Matek F405-STD

### ISSUE THREAD

**Issue by devdeven** [iNavFlight/inav-configurator#674]
Hello

I am unable to flash the Matek F405-STD with Configurator 2.1.1. It downloads the firmware online when I go to press Flash Firmware I get "USB Device successfully closed"
I am using a MacBook Pro running OSX 10.13.6

Just tested flashing with configurator 2.0.0.
Flashing successful

**Comment by shellixyz** [comment]
Could be OS related or already fixed, I can't reproduce the issue (configurator master, F405-STD)

**Comment by devdeven** [comment]
I have the same issue with Windows 10. Possibly something I am doing wrong when flashing with the new configurator?  I don't understand why the older one works fine. 
Can replicate the issue with Matek F405-OSD

**Comment by devdeven** [comment]
Are you using the standalone configurator or chrome?

**Comment by shellixyz** [comment]
I used the standalone configurator on Linux

**Comment by hnelson** [comment]
I'm unable to flash my Matek F722-STD from MacOS 10.13.6 using configurator versions 2.1.1 and 2.1.2.  

I can flash other firmware such as Betaflight, Cleanflight, and iNav 1.9.x, but not iNav 2.1.x.

**Comment by Troynica** [comment]
I observe the same or similar issue: I cannot flash any board with standalone configurator:
- v2.1.0 (MacOS)
- v2.1.1 (MacOS)
-v2.1.3 (MacOS)
using either downloaded firmware or locally produced.
When the firmware is loaded, the progress bar displays **Loaded Local Firmware: (254768 bytes)**, but nothing happens when clicking the button **Flash Firmware**. I can wait, but still nothing happens. However, in the log is updated by two lines every time the button is clicked:

```
2019-02-20 @ 17:44:12 -- USB device successfully opened with ID: 26
2019-02-20 @ 17:44:12 -- USB device successfully closed
2019-02-20 @ 17:44:13 -- USB device successfully opened with ID: 27
2019-02-20 @ 17:44:13 -- USB device successfully closed
```

My workaround is to flash the boards with version 2.0.0 and use the current to configure parameters.

**Comment by Pierre-A** [comment]
Same issue here, impossible to flash iNav 2.1.0 on an Omnibus F4Pro.
My config :
- MacBook Air (with Sierra 10.13.6)
- iNav Configurator 2.1.4 (standalone)
and same observations as above by Troynica.

My workaround was to
- load locally the v2.1.0 iNav.hex file
- then flash this file on the Omnibus F4Pro with Cleanflight Configurator 2.4.0 (selecting Omnibus F4SD board).

The board is then connected as usual to iNav configurator, and previous parameters are pasted in CLI. Everything works fine with the new iNav 2.1.0.

So, it seems that this issue is more likely related to the iNav configurator and I also suggest to rename it accordingly.

**Comment by shellixyz** [comment]
Was it working with configurator 2.0.x ?

**Comment by Pierre-A** [comment]
No, same issue with 2.0.
Just tried with same config and same procedure; DFU not detected and used USB manual selection.

**Comment by shellixyz** [comment]
Only Alberto from the devs have Apple computers running iOS I think.

**Comment by happilyb** [comment]
I have a hakrc minif4 f4 flight controller that doesn't flash in latest inav. Manage to get Omnibus Pro 2.0  version to flash with inav 2.0 on a Mac but that turned out to be wrong target anyway. Is this just a Mac issue?

**Comment by shellixyz** [comment]
Yes it looks like an iOS only issue

**Comment by happilyb** [comment]
Thanks for reply, is there a fix for this in the future ? or do I have to
steal my sons windows laptop

**Comment by shellixyz** [comment]
I don't know. I don't have any computer with iOS myself. Someone with one will have to debug and fix this.

**Comment by donjakobo** [comment]
Can also confirm, trying to flash MatekF411 board, and same cycle of "open -> close" for device, able to read current config fine, so nothing is written at all.

> 2019-04-03 @ 11:09:44 -- USB device successfully opened with ID: 6
>2019-04-03 @ 11:09:44 -- USB device successfully closed
>2019-04-03 @ 11:09:44 -- USB device successfully opened with ID: 7
>2019-04-03 @ 11:09:44 -- USB device successfully closed

**Comment by spacyje** [comment]
Same problem with MATEK F722 STD

**Comment by evmoius** [comment]
Same issue trying to flash a F405 STD.  Could not flash on OSX or Windows 10 with 2.1+ configurators.
The 2.0 configurator worked and I can connect with the 2.1 version to configure.

**Comment by happilyb** [comment]
I had no trouble with windows 10 on laptop, loaded 2.1.4 configurator, the latest drivers and installed them and I could flash the latest firmware, which I couldn’t do with Mac Sierra

**Comment by hoagietime** [comment]
I'm having the same problem with a F411-wing on a Macbook. I've tried older versions with no luck. Haven't had any trouble flashing boards on betaflight after installing whatever drivers they required.

**Comment by peteoz** [comment]
same issue with Matek F405-Wing, on MacBook Pro with standalone configurator 2.1.4, the issue first reported more than 3 months ago, is there no solution to this problem ?

**Comment by hnelson** [comment]
I think the current iNav Configurator team lacks an Mac OS person to fix
the problem.  I used an old version of the configurator to do the flash of
the new version of the configurator onto the board.  Then I used the new
configurator to talk to the board after the flash.  It's kludgy, but it
works for the time being.

Hugh

**Comment by hoagietime** [comment]
@hnelson @peteoz  I tried convigurator 2.0 and I was able to flash the board with the current version! Thanks for the tip!

**Comment by Troynica** [comment]
The following workaround works for me without having to rely on an older version of the configurator. Or any configurator at all for that matter.

These are the prerequisites:
- You have to be comfortable using the command line
- The ARM toolchain needs to be installed (I use gcc-arm-none-eabi-8-2018-q4-major, obtainable [here](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads)). For convenience, add the bin folders to your PATH
- dfu-utils need to be installed. I use Homebrew: `brew install dfu-util`

In short:
Convert the hex file to a bin file:
`<path-to>/objcopy --input-target=ihex --output-target=binary <hex file name> <bin file name>`

Upload the bin file to the board:
`dfu-util -s 0x08000000:leave -a 0 -D <path-to>/<your-bin-file>`

Above tested on F3 and F4 boards.

If this works for you, you can ditch outdated configurators if you only use those for flashing firmware.

**Comment by rmaia3d** [comment]
I'm also having this very same problem. 2010 MacbookPro running MacOS 10.13.6.

Won't flash any firmware with the 2.2.0 configurator. Even the 2.0.0 configurator won't flash also. It detects the DFU mode, opens the flasher tab, but after selecting the .hex file to be flashed and clicking "flash firmware", it just does nothing. On the events log, you can a see a message "USB connection open" and then another "USB connection closed" with the same timestamp. And nothing else.

I was able to flash the board (Matek F405-CTR) using the 1.9.3 configurator, as others have mentioned.

But I did notice that sometimes the 1.9.3 will also exhibit the same problem. From the experimenting I did, it mostly happens if you connect the board in DFU mode with the configurator already open. If you connect the board to the USB first, and only after that open the configurator (1.9.3), it will work fine.

My "workflow" for flashing the board is like this at the moment:
- with 2.2.0 configurator open, I connect the board in DFU mode.
- wait for the configurator to recognize the DFU mode.
- close the 2.2.0 configurator (command-Q).
- leave the board connected.
- with the board still connected, open the 1.9.3 configurator
- go to flasher tab and flash the firmware
- after successful flash, close the 1.9.3 configurator
- open 2.2.0 configurator and proceed to configure the board

**Comment by flyandride** [comment]
I've had the same problem with different Matek boards.  The easiest workaround is to download the firmware and use the Betaflight firmware flasher, using Load Firmware Local.  Works every time but I hope the iNav devs fix this.

**Comment by dingrgh** [comment]
I am unable to flash firmware on my Matek F405-Wing board (target Matek F405SE) with the Configurator V2.2.1. Flashes fine with Configurator 2.0.

MacOS using standalone configurator.

**Comment by PaulWieland** [comment]
Just ran into this issue with the latest version of iNav configurator and a Matek F405 Mini. Downloaded Betaflight Configurator and I can flash with it just fine, so clearly this is an issue with the iNav Configurator.

**Comment by happilyb** [comment]
I went back to an INav configurator( at that time it was 2.0) that would
flash for me and got the latest firmware from there and then opened the fc
in the latest configurator. I have had no problems since 2.3, hope you can
work it out, I am on a Mac
